src/main/scala/ylabs/messaging/Client.scala

```diff
@@ -142,10 +149,23 @@ class Client extends FSM[State, Context] {
       val accountManager = AccountManager.getInstance(connection)
       Try {
         val (username, _) = splitUserIntoNameAndDomain(register.user)
+        if(username.value == domain) throw new Messages.InvalidUserName(register.user)
         accountManager.createAccount(username.value, register.password.value)
       } match {
         case Success(s) ⇒ log.info(s"${register.user} successfully created")
-        case Failure(t) ⇒ log.error(t, s"could not register ${register.user}!")
+        case Failure(t) ⇒
+          log.error(t, s"could not register ${register.user}!")
+          val response: ActorFailure = t match {
+            case ex:Messages.InvalidUserName => ActorFailure(ex)
+            case ex:XMPPErrorException =>
```

Comment:
please run scalariform (either automatically or manually) - we use mostly the standard settings - ask @mourednik for assistence with intellij